src/common/utils/MetaKeyUtils.cpp

```diff
@@ -1190,10 +1192,16 @@ GraphSpaceID MetaKeyUtils::parseListenerSpace(folly::StringPiece rawData) {
 }
 
 PartitionID MetaKeyUtils::parseListenerPart(folly::StringPiece rawData) {
-  auto offset = kListenerTable.size() + sizeof(meta::cpp2::ListenerType) + sizeof(GraphSpaceID);
+  auto offset = kListenerTable.size() + sizeof(GraphSpaceID) + sizeof(meta::cpp2::ListenerType);
   return *reinterpret_cast<const PartitionID*>(rawData.data() + offset);
 }
 
+ListenerID MetaKeyUtils::parseListenerId(folly::StringPiece rawData) {
```

Comment:
Perhaps you need to handle the old existing data which does not have listener id.

Reply:
Good catch.  
Compatibility has already supported.